runtime/src/main/java/org/corfudb/runtime/collections/StreamingManager.java

```diff
@@ -0,0 +1,145 @@
+package org.corfudb.runtime.collections;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.exceptions.StreamSubscriptionException;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * A streaming subscription manager that allows clients to listen on
+ * the transaction updates of interested tables. The updates will be
+ * streamlined and clients can get notifications via the registered
+ * call backs.
+ * <p>
+ * Created by WenbinZhu on 11/5/20.
+ */
+@Slf4j
+public class StreamingManager {
+
+    // Number of thread in polling and notification pool.
+    private static final int NUM_THREAD_PER_POOL = 4;
```

Comment:
How did we arrive at this number ?

Reply:
This is based on an estimation of subscribers and and limits of threads.